include/highfive/bits/H5DataType_misc.hpp

```diff
@@ -151,9 +180,75 @@ inline AtomicType<std::complex<double> >::AtomicType()
     _hid = H5Tcopy(complexType.getId());
 }
 
+// Other cases not supported. Fail early with a user message
+template <typename T>
+AtomicType<T>::AtomicType() {
+    static_assert(details::array_dims<T>::value == 0,
+                  "Atomic types cant be arrays, except for char[] (fixed-len strings)");
+    static_assert(details::array_dims<T>::value > 0, "Type not supported");
+}
+
+
+
+template <std::size_t N>
+inline FixedLenStringArray<N>
+::FixedLenStringArray(const char array[][N], std::size_t length) {
+    datavec.resize(length);
+    std::memcpy(datavec[0].data(), array[0].data(), N * length);
+}
+
+template <std::size_t N>
+inline FixedLenStringArray<N>
+::FixedLenStringArray(const std::string* iter_begin, const std::string* iter_end) {
+    datavec.resize(static_cast<std::size_t>(iter_end - iter_begin));
+    for (auto& dst_array : datavec) {
+        const char* src = (iter_begin++)->c_str();
+        const size_t length = std::min(N - 1 , strlen(src));
+        std::memcpy(dst_array.data(), src, length);
+        dst_array[length] = 0;
+    }
+}
+
+template <std::size_t N>
+inline FixedLenStringArray<N>
+::FixedLenStringArray(const std::vector<std::string> & vec)
+    : FixedLenStringArray(&vec.front(), &vec.back()) {}
+
+template <std::size_t N>
+inline FixedLenStringArray<N>
+::FixedLenStringArray(const std::initializer_list<std::string>& init_list)
+    : FixedLenStringArray(init_list.begin(), init_list.end()) {}
+
+template <std::size_t N>
+inline void FixedLenStringArray<N>::push_back(const std::string& src) {
+    datavec.emplace_back();
+    const size_t length = std::min(N - 1 , src.length());
+    std::memcpy(datavec.back().data(), src.c_str(), length);
+    datavec.back()[length] = 0;
+}
+
+template <std::size_t N>
+inline std::string FixedLenStringArray<N>::operator[](std::size_t i) const {
+    return std::string(datavec[i].data());
```

Comment:
The copy in `operator[]` makes me want to cry ;)

With C++17, we could use `std::string_view`.
But the GSL already provides `string_span` that we could probably leverage.

Reply:
I created a separate issue about that, see #277